# Q3) Sales of products in four different regions is tabulated for males and females. Find if male-female buyer rations are similar across regions.

In [1]:
import pandas as pd
import numpy as np
from scipy import stats
import matplotlib as plt
import seaborn as sn

In [12]:
data = pd.read_csv("C:\\Users\\Anks\\Downloads\\BuyerRatio (1).csv")

In [13]:
data.head()

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [14]:
Males = data.iloc[0,1:]
Females = data.iloc[1,1:]


In [15]:
data1 = [Males,Females]
print(data1)

[East      50
West     142
North    131
South     70
Name: 0, dtype: object, East      435
West     1523
North    1356
South     750
Name: 1, dtype: object]


In [16]:
result = stats.chi2_contingency(data1)
print(result)

(1.595945538661058, 0.6603094907091882, 3, array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
       [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))


In [17]:
stat_value = result[0]
pi_value = result[1]
df = result[2]
stat_value, pi_value, df

(1.595945538661058, 0.6603094907091882, 3)

In [18]:
expected_value = result[3]
expected_value

array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
       [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]])

Ans) As p-value(0.6603) is greater than alpha significance(0.05) we can conclude that - "We fail to reject null hypothesis i.e. All proportions are equal

# Q4) TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain % of the customer order forms. Any error in order form renders it defective and has to be reworked before processing. The manager wants to check whether the defective % varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences

In [21]:
data = pd.read_csv("C:\\Users\\Anks\\Downloads\\Costomer+OrderForm.csv")

In [22]:
data.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

In [23]:
phillippines = data.Phillippines.value_counts()
indonesia = data.Indonesia.value_counts()
malta = data.Malta.value_counts()
india = data.India.value_counts()


In [24]:
phillippines.values
indonesia.values
malta.values
india.values

array([280,  20], dtype=int64)

In [25]:
df_1 = [phillippines[0],indonesia[0],malta[0],india[0]]
df_2 = [phillippines[1],indonesia[1],malta[1],india[1]]
df = [df_1,df_2]
df

[[271, 267, 269, 280], [29, 33, 31, 20]]

In [26]:
result = stats.chi2_contingency(df)
result

(3.858960685820355,
 0.2771020991233135,
 3,
 array([[271.75, 271.75, 271.75, 271.75],
        [ 28.25,  28.25,  28.25,  28.25]]))

In [27]:
stat_value = result[0]
pi_value = result[1]
df = result[2]
stat_value, pi_value, df

(3.858960685820355, 0.2771020991233135, 3)

Ans) As p-value(0.2771) is greater than alpha significance(0.05) we can conclude that - "We fail to reject null hypothesis i.e. All proportions are equal."